### ISSUE TITLE
Startup commands not running cause continuing a tutorial to not work

### ISSUE THREAD

**Issue by moT01** [freeCodeCamp/CodeAlly-CodeRoad-freeCodeCamp#39]
Related to [this comment](https://github.com/coderoad/coderoad-vscode/issues/558#issuecomment-1005746018)

The startup commands often change things in the system. For example, postgres config files, bashrc commands - these are for generating logs - and other things like starting the postgres server. When closing a container and trying to continue a tutorial, the workspace files seem to remain, but these other things go away - sort of. I believe when running locally in Docker, it keeps the system files (postgres config, bashrc, etc.), but the postgres server shuts down. In CodeAlly, I believe it removes the system files and stops the services.

I can somewhat mitigate this within the tutorial by moving certain commands to the reset button. So when continuing, users would have to hit the reset button to copy the files back over to the system and start the service back up. I could do that, and it would be good enough for now if it needs to be. It does makes users take an extra step to continue, and it's not super clear that they need to take that step. I will probably write up, or have something to help users out. Maybe a FAQ page or something where I could put info like this.

We could also run the startup commands when continuing. I believe CodeRoad used to do this. Maybe that would be the easiest, and I think it would work. Although, some commands that do change workspace stuff, like `npm install` to install modules, would not need to be run again. So maybe a separate set of commands for continuing or something? Or maybe just running the startup commands would be better. Curious if you have any thoughts on this one @ShMcK.

Side note: I was thinking about including the modules in the repo so they wouldn't need to be installed after loading it

**Comment by AdamZaczek** [comment]
It's as you wrote @moT01. Only some folders are stored across docker sessions.
The more folders we store, the bigger the data usage. It could quickly go out of control with the amount of expected traffic.

We store folders such as 
```
~/project/
~/.local/
```
so they can be used for small stuff. Some packages, such as PostgreSQL-related ones, are hard to use in unusual destinations though.

Running commands that will sync the remaining folders sounds like the best approach.

Let me know if we can help somehow.

**Comment by ShMcK** [comment]
I'll add a continue hook that will let you run any commands or vscodeCommands.

```
tutorial.config.continue: { commands: [], vscodeCommands: [] }
```

I'll also a webhook if needed to track continued tutorials.

```
tutorial.webhook.events.continue
```

PR: https://github.com/coderoad/coderoad-vscode/pull/561

Released in v0.19.0

**Comment by moT01** [comment]
This is workin real good. I added continue commands to all the tutorials.
